### DISCUSSION TITLE
Is there any way to catch the Vector instance Crash case

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#15540] [Q&A] [answered]
Hello Team,
We have a case where want to introduce the Service level redundancy in our system. We wanted to switch to the redundant server's vector network as soon as there is a crash in the primary server's vector network. In this case we wanted to understand if there is a way to figure out to catch the crash case of vector instance, which would help us to switch from primary to secondary.

**Comment by jszwedko** [comment, marked as answer]
Hi @hemanthofficial3009 ! Some ideas:

* Run Vector with the API exposed and monitor for it being inaccessible
* Send Vector's internal metrics to an external service and monitor the `build_info` metric to see if it ceases to be reported by a given host
